src/mono/mono/utils/mono-dl.c

```diff
@@ -168,6 +171,37 @@ fix_libc_name (const char *name)
 }
 #endif
 
+/**
+ * mono_dl_open_self:
+ * \param error_msg pointer for error message on failure
+ *
+ * Returns a handle to the main program, on android x86 it's not possible to 
+ * call dl_open(null), it returns a null handle, so this function returns RTLD_DEFAULT
+ * handle in this platform.
+ */
+MonoDl*
+mono_dl_open_self (char **error_msg)
+{
+#if defined(ENABLE_NETCORE) && defined(TARGET_ANDROID) && !defined(TARGET_ARM64) && !defined(TARGET_AMD64)
```

Comment:
Why is this netcore-only?

Reply:
Because dlfcn.h is not avaiable when compiling in mono/mono repo, the non-netcore mono. Maybe something with NDK version or cross compiler, we are not sure, but we decided to fix using ENABLE_NETCORE.